scripts/install/nextcloud.sh

```diff
@@ -13,218 +13,166 @@
-        try_files \$uri/ =404;
-        index index.php;
-    }
-
-    # Adding the cache control header for js, css and map files
-    # Make sure it is BELOW the PHP block
-    location ~ ^\/nextcloud\/.+[^\/]\.(?:css|js|woff2?|svg|gif|map)\$ {
-        try_files \$uri /nextcloud/index.php\$request_uri;
-        add_header Cache-Control "public, max-age=15778463";
-        # Add headers to serve security related headers  (It is intended
-        # to have those duplicated to the ones above)
-        # Before enabling Strict-Transport-Security headers please read
-        # into this topic first.
-        #add_header Strict-Transport-Security "max-age=15768000; includeSubDomains; preload;" always;
-        #
-        # WARNING: Only add the preload option once you read about
-        # the consequences in https://hstspreload.org/. This option
-        # will add the domain to a hardcoded list that is shipped
-        # in all major browsers and getting removed from this list
-        # could take several months.
-        add_header Referrer-Policy "no-referrer" always;
-        add_header X-Content-Type-Options "nosniff" always;
-        add_header X-Download-Options "noopen" always;
-        add_header X-Frame-Options "SAMEORIGIN" always;
-        add_header X-Permitted-Cross-Domain-Policies "none" always;
-        add_header X-Robots-Tag "none" always;
-        add_header X-XSS-Protection "1; mode=block" always;
-
-        # Optional: Don't log access to assets
-        access_log off;
-    }
-
-    location ~ ^\/nextcloud\/.+[^\/]\.(?:png|html|ttf|ico|jpg|jpeg|bcmap)\$ {
-        try_files \$uri /nextcloud/index.php\$request_uri;
-        # Optional: Don't log access to other assets
-        access_log off;
-    }
+function _bootstrap() {
+    echo_progress_start "Configuring Nextcloud settings"
+
+    _occ "maintenance:install -n --database 'mysql' --database-name 'nextcloud'  --database-user 'nextcloud' --database-pass '$nextcldMySqlPW' --admin-user '$masteruser' --admin-pass '$masterpass' "
+    _occ "maintenance:mode --on"
+    _occ "db:add-missing-indices"
+    _occ "db:convert-filecache-bigint --no-interaction"
+    # _occ "config:system:set memcache.local --value='\OC\Memcache\APCu" # TODO follow https://github.com/nextcloud/server/issues/24567
+
+    i=1
+    _occ "config:system:set trusted_domains $i --value='localhost'"
+    ((i++))
+    ip=$(ip route get 1 | sed -n 's/^.*src \([0-9.]*\) .*$/\1/p')
+    _occ "config:system:set trusted_domains $i --value=$ip"
+    ((i++))
+    _occ "config:system:set trusted_domains $i --value=$(hostname)"
+    ((i++))
+    _occ "config:system:set trusted_domains $i --value=$(hostname)"
+    for value in $(grep server_name /etc/nginx/sites-enabled/default | cut -d' ' -f 4 | cut -d\; -f 1); do
+        if [[ $value != "_" ]]; then
+            _occ "config:system:set trusted_domains $i --value=$value"
+            ((i++))
+        fi
+    done
```

Comment:
Does the let's encrypt hook need to be updated?

Reply:
what do you mean by updating the LE hook?

Reply:
we only have vsftpdfd and znc hooks, not a swizzin one